gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/service/impl/UserServiceImpl.java

```diff
@@ -239,6 +239,16 @@ public Single<User> create(User user, String idp, String baseUrl, io.gravitee.am
                                                 .switchIfEmpty(Single.error(() -> new UserProviderNotFoundException(source)))
                                                 .flatMap(userProvider -> userProvider.create(UserMapper.convert(userModel)));
                                     })
+                                    .flatMap(idUser ->
+                                            userRepository.findByExternalIdAndSource(ReferenceType.DOMAIN, domain.getId(), idUser.getId(), source)
+                                                    .isEmpty()
+                                                    .map(isEmpty -> {
+                                                        if (FALSE.equals(isEmpty)) {
+                                                            throw new UniquenessException("User with externalId [" + idUser.getId() + "] already exists");
+                                                        }
+                                                        return idUser;
+                                                    })
+                                    )
```

Comment:
**issue**: We should do it before the checkRoles (line 228) as at this stage, we may have created the user into the UserProvider (line 240)